crates/egui_demo_app/src/apps/custom3d_glow.rs

```diff
@@ -84,44 +84,72 @@ impl RotatingTriangle {
     fn new(gl: &glow::Context) -> Self {
         use glow::HasContext as _;
 
-        let shader_version = if cfg!(target_arch = "wasm32") {
-            "#version 300 es"
-        } else {
-            "#version 330"
-        };
+        let shader_version = egui_glow::ShaderVersion::get(gl);
 
         unsafe {
             let program = gl.create_program().expect("Cannot create program");
 
-            let (vertex_shader_source, fragment_shader_source) = (
-                r#"
-                    const vec2 verts[3] = vec2[3](
-                        vec2(0.0, 1.0),
-                        vec2(-1.0, -1.0),
-                        vec2(1.0, -1.0)
-                    );
-                    const vec4 colors[3] = vec4[3](
-                        vec4(1.0, 0.0, 0.0, 1.0),
-                        vec4(0.0, 1.0, 0.0, 1.0),
-                        vec4(0.0, 0.0, 1.0, 1.0)
-                    );
-                    out vec4 v_color;
-                    uniform float u_angle;
-                    void main() {
-                        v_color = colors[gl_VertexID];
-                        gl_Position = vec4(verts[gl_VertexID], 0.0, 1.0);
-                        gl_Position.x *= cos(u_angle);
-                    }
-                "#,
-                r#"
-                    precision mediump float;
-                    in vec4 v_color;
-                    out vec4 out_color;
-                    void main() {
-                        out_color = v_color;
-                    }
-                "#,
-            );
+            let (vertex_shader_source, fragment_shader_source) =
+                if shader_version.is_new_shader_interface() {
+                    (
+                        r#"
+                        const vec2 verts[3] = vec2[3](
+                            vec2(0.0, 1.0),
+                            vec2(-1.0, -1.0),
+                            vec2(1.0, -1.0)
+                        );
```

Comment:
Just a side note: I think, it will be more readable, if put shaders to the separate file and `include_str!`  here.